geniushubclient/zoneclasses/properties.py

```diff
@@ -0,0 +1,43 @@
+from geniushubclient.const import (
+    ITYPE_TO_TYPE,
+    ZONE_KIT,
+    ZONE_TYPE,
+)
+
+
+class Properties:
+    """The properties of the zone.
+
+        Intended to be the properties that are fixed for the lifecycle of the zone
+    """
+
+    id: int = 0
+    name: str = ""
+    zone_type: int = 0
+    subtype: int = 0
+    type_name: str = ""
+    has_room_sensor: bool = False
```

Comment:
Why add these as class attrs? Are they needed? Why can't we just have the instance attrs? If they are needed, why can't they be constants (UPPERCASE)?

Reply:
Sorry, that's down to me not knowing the differences between C# and Python! In C#, you declare all attributes on the class like that. What you call class attributes are known as static variables, so when you declare them on the class, they just have the static keyword added to them.

I'll remove them, but I must admit that I find it frustrating! I put them there so that I could put the type on them, so that anyone new to the project would at least know what type each variable is meant to be (as it's something that I struggled with initially). You lose that metadata. Oh well, I guess that's the fun you get with duck-typed languages! :)

Reply:
Actually, I need to have them in the init method, don't I? So if I set them up in there with default values, that'd be alright, wouldn't it?

Reply:
You can have the typing in the method, no problem (and to be encouraged) - it is not 'needed' but is essentially documentation (unless you take additional steps):
```
subtype: int = None
type_name: str = None
```

Only provide a default value if there *is* a default value - this is distinct from an initial value, if it is either invalid (e.g. name = ""), or incorrect (subtype = 0).  

Here is  better example:

If you had a class for a **Soccer** game, then an initial value of `score = 0` is appropriate, but if it is for **Weather**, then a `temperature = 0` is not & it should be either:
```
Class Weather(self):
    self.temperature: float = None
```
... or:
```
Class Weather(self, temp)
    self.temperature: float = temp
```